Generalize book

spirosChv opened the pull request:
1. [toc](https://github.com/NeuromatchAcademy/nmaci/blob/generalize_book/scripts/generate_book.py#L15-L21) dictionary is now not hardcoded

2. add [repo variable](https://github.com/NeuromatchAcademy/nmaci/blob/generalize_book/scripts/generate_book.py#L9)
3. projects: the chapters are [based on the course repo](https://github.com/NeuromatchAcademy/nmaci/blob/generalize_book/scripts/generate_book.py#L59)
4. [preprocess intro.ipynb if it exists](https://github.com/NeuromatchAcademy/nmaci/blob/generalize_book/scripts/generate_book.py#L114)

spirosChv commented:
- [ ] We have to make the project chapter, the schedule, and the technical chapters more abstract. I suggest adding the dictionary structures as an input parameter read from a txt file located in each repository. Now, I have made the course-content-dl similar to course-content.

ebatty commented:
Looks good! We will need to add the art section I just changed in the main version

spirosChv commented:
> Looks good! We will need to add the art section I just changed in the main version

@ebatty I have added the art section. Please check if this is ok.
